### PR TITLE
Refactor tests for property queries and aggregate queries

### PR DESCRIPTION
**Pull request by kevinwcyu**
Refactors the test for property value and property value aggregate queries.

Fixes: #188

Changes are based on comments from https://github.com/grafana/iot-sitewise-datasource/pull/179#discussion_r1190933637.

Changes made:
- Update tests names
- Keep inputs/outputs closer to the test case
- Add assertions to make sure all the mocks are called
- Add assertions on the important arguments to the mocks

Notes:
- Tests still need to be updated for other test files. I've created issue here https://github.com/grafana/iot-sitewise-datasource/issues/189. Just a placeholder for now. It's probably easier to tackle all of these in smaller PRs.

### CODE REVIEW

**Comment by fridgepoet** [comment]
Yowza! Nice job, looked like a lot of work. Pretty overwhelming to read still but I'm not sure what could make an end-to-end style test more readable. Another idea I have (future?) is to have smaller tests at each layer. The backend in this repo is a little bit "too much" but at least there is a method to the madness, so there are some distinct layers...

I hope, though, that your change will make some debugging easier one day.

Something I am missing is where the golden files are still getting modified. I see they were removed in these files you modified, but do you think there was some value to having the large sets of data? (I don't actually know if there were large sets in these tests.) And also were there any files we are no longer using now that we pulled some input into the test files? 

> Update Property method in pkg/resource/sitewise.go so it will call DescribeAssetPropertyWithContext as long as an asset id and property id is available. Previously, it would not call DescribeAssetPropertyWithContext if propertyAlias was not an empty string, but now that we get the asset id and property id from the alias, we can call DescribeAssetPropertyWithContext.

I think we should make this the focus of this PR if it's a change in the behavior, or at least add it as one of the main things the PR does. What do you think?

File: pkg/resource/sitewise.go
Comment (sarahzinger) [inline pkg/resource/sitewise.go:31]: Might be good to separate out changes to the codebase from refactoring tests wdyt?

File: pkg/server/test/property_value_aggregate_test.go
Comment (sarahzinger) [inline pkg/server/test/property_value_aggregate_test.go:38]: do you think it would be easier to read if we named some of these inputs? I'm not sure if this is stylistically weird to do with go but I guess it feels like it'd be easier to follow this test if we were like:
```
mockSw.On(
		"BatchGetAssetPropertyAggregatesPageAggregation",
		mockRequestContext,
		mockInput,
		mockPages,
		mockMaxResults,
	)
```

wdyt? Maybe I'm being silly/nitpicky?

File: pkg/server/test/property_value_test.go
Comment (sarahzinger) [inline pkg/server/test/property_value_test.go:85]: sorry for the very basic questions on my part but like what does this assert?

**Review by sarahzinger — APPROVED** [review]
This looks cool to me but I'm also not very confident in my golang skills yet, so I defer to @fridgepoet on this one! Thanks for taking this on!

File: pkg/server/test/property_value_test.go
Comment (fridgepoet) [reply pkg/server/test/property_value_test.go:85]: It makes sure that all the all the declared methods ("mockSw.On("...")") were called.

Sometimes we declare those mocks but the test doesn't even use them which can be misleading. It's kind of an indirect way to assert some paths/coverage, I guess. 
I think some people might put these in everywhere, but I'm not sure it's always worth it. The reason I suggested it in these tests is that they're kind of "end to end," very long, complex, repetitive, and there are many to re-do. While pairing there were a few times we noticed that a test was written incorrectly and not testing what we thought.

File: pkg/server/test/property_value_aggregate_test.go
Comment (fridgepoet) [reply pkg/server/test/property_value_aggregate_test.go:38]: These are just really huge tests since they're at the highest level (feels a bit end to end); I cannot yet find the ideal balance between something "digestable" and something where the reader doesn't get re-directed all the time to find the underlying values. 

It's my opinion that by not extracting the inputs into variables, we make it easier to connect the query input to what's being passed to some of the methods, and so on down the chain.

I think this test remains difficult to read, but mostly because is testing a very long chain of external API calls. Some ideas: maybe duplicate this test to only make assertions on one method at a time (so we can just use mock.Anything) or we could separate the tests and test at lower levels.

I _think_ this test a reasonable improvement over the previous input which was read from very far away files and sometimes quite lengthy, and an output which was also written to some file far away. I wouldn't be surprised to see further improvements iteratively as we dive into this repo more. 

Anyway, I will ultimately support whatever Kevin thinks is best!

File: pkg/resource/sitewise.go
Comment (kevinwcyu) [reply pkg/resource/sitewise.go]: Extracted changes here #192. Once that one is merged, I'll merge this PR. Temporarily switched the base branch from `main` for this PR to #192 so this only shows the refactoring of the tests.

File: pkg/server/test/property_value_aggregate_test.go
Comment (kevinwcyu) [reply pkg/server/test/property_value_aggregate_test.go:38]: i think i'll leave it as is for now and we can iterate on further improvements later.

i think keeping it as `mock.Anything` has the advantage of being more clear since it's what's provided by the mocking library and also when I hover over it I see the following so i can see a particular argument isn't important for the given test case.

```
const mock.Anything untyped string = "mock.Anything"
Anything is used in Diff and Assert when the argument being tested shouldn't be taken into consideration.
```

also a more minor reason, if we ever modify the arguments to these functions, we'd need to check every single place to make sure name still makes sense.

**Review by iwysiu — APPROVED** [review]
Nice work! It is easier to read without having to cross reference golden files

**Pull request merged**
